test/inductor/test_aot_inductor.py

```diff
@@ -3401,23 +3400,21 @@ def forward(self, a, b):
 
             example_inputs = (torch.randn(100), torch.tensor(10))
             ep = draft_export(M(), example_inputs)
+            report = ep._report
+            need_config_patch = any(
+                not f.xfail and f.failure_type == FailureType.MISMATCHED_FAKE_KERNEL
+                for f in report.failures
+            )
             m = ep.module()
-            from torch.fx.passes.fake_tensor_prop import FakeTensorProp
+            from torch._functorch import config as functorch_config
 
-            example_inputs = [
-                node.meta["val"] for node in m.graph.nodes if node.op == "placeholder"
-            ]
-            fake_mode = example_inputs[0].fake_mode
-            with enable_python_dispatcher(), fake_mode:
-                FakeTensorProp(m, mode=fake_mode).propagate_dont_convert_inputs(
-                    *example_inputs
-                )
-
-            # TODO: change to the tests below after  MetadataMismatchError is fixed
-            # pt2_file = torch._inductor.aoti_compile_and_package(ep)
-            # optimized = torch._inductor.aoti_load_package(pt2_file)
+            with functorch_config.patch(
+                {"generate_fake_kernels_from_real_mismatches": need_config_patch}
+            ):
```

Comment:
can you write a TODO  that this should no longer be needed after https://github.com/pytorch/pytorch/pull/150093